components/chat.tsx

```diff
@@ -30,6 +39,133 @@ export function Chat({
   isReadonly: boolean;
 }) {
   const { mutate } = useSWRConfig();
+  const [swapDetails, setSwapDetails] = useState<SwapTransactionDetails | null>(
+    null
+  );
+
+  // Handle wallet connection and transaction
+  useEffect(() => {
+    console.log("@@swapDetails", swapDetails);
+    if (swapDetails) {
+      console.log("swapDetails", swapDetails);
+      const handleTransaction = async () => {
+        try {
+          console.log("Starting the provider");
+          // Get Phantom provider
+          const provider = (window as any).phantom?.solana;
+          if (!provider?.isPhantom) {
+            console.log("Phantom provider not found");
+            window.open("https://phantom.app/", "_blank");
+            return;
+          }
+
+          console.log("Connecting to wallet");
+          // Connect to wallet
+          await provider.connect();
+          console.log("Connected to wallet");
+
+          // Create and send transaction
+          // Use a more reliable RPC endpoint
+          const connection = new Connection(
+            "https://mainnet.helius-rpc.com/?api-key=b61a3f14-fb2e-49fc-9f3f-7f7b0b6a3a0c",
```

Comment:
It's bad here but as it's frontend app seems the best is to make the call from the backend?

Reply:
What's the risk of someone stealing this key? Do we pay for it? 
If no major risk and this is for demo purposes we can leave it i guess. But for prod systems, definitely it should go through the backend proxy.